### ISSUE TITLE
Change LICENSE

### ISSUE THREAD

**Issue by non-binary** [Quick/Quick#660]
Hey everyone. I'd like to start a discussion about something that may become controversial or "problematic." All I ask is that we abide our CoC and keep it civil. ❤️  I'd like to propose that we relicense Quick and Nimble to use something other than the Apache licenses.

## Why?

By using the Apache license, we're complicit in cultural appropriation. The name Apache originates from the Native American tribe: [Apache](https://en.wikipedia.org/wiki/Apache). From my somewhat thorough release, I couldn't find anything that says that [Apache Software Foundation](https://www.apache.org/) received permission to use the name.

We work in an industry which makes less privileged people feel like the tech industry is only for 
abled cis straight white men. I have done my best, as well as many others have to help change this perception and this is one more thing I believe we need to do. I'm tired of seeing open source projects named after sacred indigenous animals, beliefs or people but we have to start with the elephant in the room. Millions of projects on GitHub are licensed under one of the Apache licenses and we should dock two projects from that number. Quick and Nimble, and urge many other projects to do the same.

## Well, which license should we use?

I'm thinking about contacting a lawyer of sorts and having a new license written that is similar to the Apache license but minus the cultural appropriation.

However, what's everyone's thoughts on making Quick and Nimble licensed under MIT?

@Quick/core; I'm really interested in seeing what everyone thinks.

Cheers!

**Comment by modocache** [comment]
That makes sense to me. I don't have a strong feeling about which license to use, nor do I know a ton about the process of changing licenses. So this will be difficult for me to take on.

One thing to keep in mind: I know of several large tech companies that use Quick. I feel these are valuable "customers," and I'd like to keep them. Let's make sure any license changes we make don't make this software any harder to use.

**Comment by istx25** [comment]
I think the process is pretty simple if we're changing the license to something more permissive like MIT. I also do not want to lose any "customers" but we can also use Quick to better the community since we have quite a presence. MIT is our best bet, imo.

**Comment by jeffh** [comment]
While I understand a the concern, I have some disagreements with MIT and BSD in the actual protections. They are not formal legalese documents that can cause some concern in regard to patent trolls that roughly goes like this:

- submit PR with patented algorithm
- wait for users to use library
- sue all users of the library

The Apache license does cover this case by forcing a patent use grant for contributed code. In this case, Apache licensed code is safer to use than MIT or BSD licensed code.

Of course, if there is another license that provides the same insurances, I'm fine with that.

Just my 2 cents.

- Jeff

**Comment by istx25** [comment]
You bring up a fair point, @jeffh. That could be a big problem. I'm looking into talking to a software license lawyer and see if we can get a new license made that provides similar or the same protections as the Apache license. I'll keep everyone up to date on my progress here.

**Comment by jeffh** [comment]
I've been thinking about this a bit more. If we go with a custom license, there are other tradeoffs. We'll need to promote the details of the license and probably have to make analogies to other licenses (e.g. - this license is Apache-like). Which leaves some questions:

- Is mentioning Apache-like still too much negative perception?
- How do we calm fears from businesses / developers of custom licenses? As a developer, I'm generally suspicious of any custom licenses. Some businesses refuse to accept any OSS that's not on a pre-approved license list.

These are not show-stoppers. But it's something to consider as your pursue a custom license.

**Comment by modocache** [comment]
> As a developer, I'm generally suspicious of any custom licenses. Some businesses refuse to accept any OSS that's not on a pre-approved license list.

Yup, this is a clearer way of putting my initial concerns. It's all about those lists. If a license isn't on there then that means developers need to call their company's legal team, which can then snowball into a huge thing. If it were me, I wouldn't go through all that hassle just to use Quick as a dependency.

We have a few large tech companies as consumers. I don't want our users to start an A/C-priv email thread because one of their dependencies adopted a new license.

As a result I'd greatly prefer if we use an existing license.

**Comment by istx25** [comment]
These are all fair concerns. Are there any licenses that are similar to the Apache license but isn't appropriating? I'm still reading and learning about the various software licenses so I may have missed some. Does anyone know of any good software license lawyers? I don't mind paying to get these questions answered by a professional. I appreciate everyones input.

**Comment by jeffh** [comment]
I don’t know any other licenses that aren’t Apache + copyleft behavior which makes it unusable in iOS context.

I’ve filed an issue to Apache Legal Discuss about this question.

https://issues.apache.org/jira/browse/LEGAL-284

Feel free to elaborate on that issue as well.

- Jeff

**Comment by istx25** [comment]
Thank you for reaching out to them but I find their response a little ridiculous. You specifically asked "_So I ask, does the Apache Foundation use "Apache" with permission from the appropriate Native American Tribe(s)?_" and they linked to https://www.apache.org/foundation/faq.html#name.

Which states...

>The name 'Apache' was chosen from respect for the various Native American nations collectively referred to as Apache, well-known for their superior skills in warfare strategy and their inexhaustible endurance. It also makes a cute pun on "a patchy web server" -- a server made from a series of patches -- but this was not its origin. The group of developers who released this new software soon started to call themselves the "Apache Group".

No where does that paragraph say they received permission or Apache's blessing to use their name. Would you be able to inquire further about that specifically? If they received permission or have any Apache people in their team leadership that were involved in the decision making when they chose the name Apache Software Foundation.

**Comment by istx25** [comment]
I didn't know I could sign up on their Jira without an invite. Just commented a version of this on your issue, @jeffh. Let's see where this goes. Thanks! :)

**Comment by jeffh** [comment]
No problem. I'll let you take it from here. Keep us updated on any new findings you arrive at. 👍

**Comment by lferro9000** [comment]
Apart from the protections of the license that the projects are and will be, the issue arised is with relation of the "Apache" word in the context of the "American Native tribes".

The first question is if ASF aka Apache License has authorization for the use of the term.

This implies that there is someone or some entity that can give such authorization (be it free or for a fee). Is that the case?

Because if there isn't, then there is no way to persue with success any kind of arrangement for a continuation of the use of the term in a way to aplease that requirement (apart from changing to a different license).

If there is, one can always ask for permission as an individual project if needed be, and comply with whatever requirements that authorization will require.

**Comment by jeffh** [comment]
That's a fair point @lferro9000, I'm not too familiar with all the legalities and nuances of American Tribe usages.

On my very brief exploration. Tribe name usages are complex – being intertwined with trademarks and special-case law. But it seems like the following applies:

- If the tribe has filed for a trademark, they can fully enforce all trademark usages. An example of this Chrysler using Cherokee for their vehicles by acquiring a trademark license from the Cherokee Tribe.
- If the tribe has not filed for a trademark, trademarks with that name fall under regulation that depends on the product's relation with that tribe. Fair-use is managed by the US Government's Bureau of Indian Affairs / Federal Trust Obligation to American Indians.
   - The Bureau's policy seems to generally as follows: **usage is acceptable if the name has some direct relationship with Native American Tribe that the name uses**. This is to prevent fraudulent usages that would indicate a product was made by a Native American Tribe when is really wasn't. For example, the **Apache Helicopter** uses parts manufactured by the **Apache Aerospace Company**, which is operated by the *White Mountain Apache Tribe*. In this context, it would considered legal.
- A tribe that registers for a trademark can *retroactively override existing trademarks* that conflict with a tribe's trademark registration. Seems like a pretty unique feature for Native American Tribes. But I digress.

It gets more complicated *specifically* with Apache:

 - Apache was the term used by the French to categorize many Native American Tribes. Those tribes do not internally associate themselves as Apache.
 - There are 9 officially registered tribes with "Apache" in their name. What's more interesting is that they tend to *not* call themselves Apache. For example, the White Mountain Apache Tribe call themselves ["Dagot'ee"](http://www.wmat.nsn.us/). The former name is probably to identify themselves to the larger English speakers around them.
 - Although there are 9 officially registered "Apache" tribes, they have been unable to acquire the "Apache" name as a trademark since the PTO requires the trademark to directly link to a specific tribe, which "Apache" does not fall under.
 - Since there's no "Apache" tribe, trademark license is harder to file for as a trademark, and can be argued that since it's not directly related to a specific tribe. I'm guessing it could be argued that the Bureau of Indian Affairs may not be able to fight a trademark using Apache.


There seems to be a past precedence that the US government has rejected trademarking of **APACHE**. There's a [PTO response to an attempt to acquire the APACHE trademark by a Native American Tribe](https://www.uspto.gov/web/offices/com/sol/foia/ttab/2aissues/2004/78175476.pdf) - TLDR: Apache isn't specific enough to a particular tribe.

In short:

- There is **always a legal entity** if it is related to a trademark of a tribe name (either the tribe, or the US Government).
- Although many tribes have Apache in its name, 'Apache' is not trademark-able by Native Americans because it refers to many tribes.
- ∴ it's likely the ASF has not acquired permission from tribe(s) because there's no legal entity that represents Apache.

But I'm also no lawyer. YMMV. Just some more info for @istx25.

**Comment by jeffh** [comment]
Hey @istx25, just checking in if you have any updates 😄

**Comment by eenblam** [comment]
Found the JIRA issue while researching a similar concern, then ended up here. I think it would be generally useful to the open source community to take Marvin Humphrey up on his advice to simply **copy the text of the license**. Whatever it was renamed to is a bit arbitrary from my point of view, assuming the issue of appropriation is resolved.

If dual licensing isn't okay'd for this project by Quick copyright holders, I would nonetheless be interested in collaborating with some folks to host a page for an alternative license. If this gets out of scope for Quick's issue tracker, feel free to open an issue on [eenblam/misc](https://github.com/eenblam/misc) for further discussion.

**Comment by namurphy** [comment]
I happened upon this issue after a web search, and a new alternative is the [BSD+Patent license](https://opensource.org/licenses/BSDplusPatent) that was approved very recently by the Open Source Initiative.  This new license consists of the BSD 2-clause license plus an explicit patent grant that protects users against software patents, though without the retaliation clause.  This is my new favorite permissive license.

I fully agree that the use of the Apache license constitutes cultural appropriation, and that it violates the first bullet point of the [code of conduct](https://github.com/Quick/Quick/blob/master/CODE_OF_CONDUCT.md) on "Using welcoming and inclusive language".

Thank you for raising this issue.
-Nick

**Comment by non-binary** [comment]
It's been a couple years since I posted on this issue. I don't contribute to Quick anymore but I just saw your response @namurphy and wanted to chime in. I think it would be pretty awesome if Quick adopted the BSD+Patent license and since it is on the approved list, I think that solves our concerns. Since I don't contribute anymore it's not up to me, but what's people's thoughts?

**Comment by non-binary** [comment]
It's been a couple years since I posted on this issue. I don't contribute to Quick anymore but I just saw your response @namurphy and wanted to chime in. I think it would be pretty awesome if Quick adopted the BSD+Patent license and since it is on the approved list, I think that solves our concerns. Since I don't contribute anymore it's not up to me, but what's y'all thoughts? @jeffh @modocache
